### PR TITLE
Add Ctrl+Backspace support

### LINKED ISSUE #755: Ctrl-backspace does not delete back to the previous wordbreak
It deletes character-by-character instead.

### PR DESCRIPTION
**Pull request by mkitzan**
## Summary of the Pull Request
Changes the <kbd>Ctrl+Backspace</kbd> input sequence and how it is processed by `InputStateMachineEngine`. Now <kbd>Ctrl+Backspace</kbd> deletes a whole word at a time (tested on WSL, CMD, and PS).

 
## PR Checklist
* [x] Closes #755
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed -> made minor edits to tests
* [ ] Requires documentation to be updated
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #755

## Detailed Description of the Pull Request / Additional comments
Changed the input sequence for <kbd>Ctrl+Backspace</kbd> to `\x1b\x8` so the sequence would pass through `_DoControlCharacter`. Changed `_DoControlCharacter` to process `\b` in a way which forms the correct `INPUT_RECORD`s to delete whole words.

## Validation Steps Performed
<kbd>Ctrl+Backspace</kbd> works 🎉

### CODE REVIEW

**Comment by mkitzan** [comment]
Thanks for the review @zadjii-msft. That's a lot of good info. I'll work on making `InputStateMachine` process <kbd>Ctrl+h</kbd>as <kbd>Ctrl+Bksp</kbd>.

**Comment by mkitzan** [comment]
With some new changes making `InputStateMachine` process `^H` as <kbd>Ctrl+Bksp</kbd>, CMD and PS will delete whole words. In WSL, `^H` does not follow the same path through  `InputStateMachine` as in CMD or PS (apparently). Once I figure out how to send WSL the correct `INPUT_RECORD`, we should be in business.

**Comment by mkitzan** [comment]
@zadjii-msft WSL standalone doesn't delete whole words when <kbd>Ctrl+Bksp</kbd> is pressed, which explains why I was having my [problems above](https://github.com/microsoft/terminal/pull/3935#issuecomment-566768716). After testing the behavior of `^H` in `showkey -a` and `conechokey`, I believe this solves the problem without messing with the VT sequences like my last version. Thanks for pointing out those command line tools btw, very helpful.

File: src/terminal/parser/ut_parser/InputEngineTest.cpp [hunk 1]
```diff
@@ -352,6 +352,9 @@ void InputEngineTest::C0Test()
         case L'\t': // Tab
             writeCtrl = false;
             break;
+        case L'\b': // backspace
+            wch = '\x7f';
+            expectedWch = '\x7f';
```
Comment (zadjii-msft) [inline src/terminal/parser/ut_parser/InputEngineTest.cpp]: nit: this should probably have a break too, even though it's the last case

Comment (mkitzan) [reply src/terminal/parser/ut_parser/InputEngineTest.cpp]: Makes sense, I included it in the latest commit

Comment (mkitzan) [reply src/terminal/parser/ut_parser/InputEngineTest.cpp]: Makes sense, the latest commit added the break statement

**Comment by zadjii-msft** [comment]
@msftbot merge this in 168 hours

**Comment by zadjii-msft** [comment]
@msftbot forget everything I just told you

**Comment by mkitzan** [comment]
Glad to have this one merged! Thanks

**Comment by Pytal** [comment]
Is it possible to get this working with git bash?

**Comment by DHowett-MSFT** [comment]
You’ll want to look into `inputrc`

**Comment by wmertens** [comment]
I'm on 0.10.781.0 but ctrl-backspace simply seems to send ctrl-h in WSL2 :(

**Comment by zadjii-msft** [comment]
@wmertens You'll want to read #4397

**Comment by DHowett-MSFT** [comment]
:tada: Once again, thanks for the contribution!

This pull request was included in a set of conhost changes that was just
released with [Windows Insider Build 19603](https://blogs.windows.com/windowsexperience/2020/04/08/announcing-windows-10-insider-preview-build-19603/).

**Pull request merged**